pybossa/api/project_locks.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf8 -*-
+# This file is part of PYBOSSA.
+#
+# Copyright (C) 2017 Scifabric LTD.
+#
+# PYBOSSA is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# PYBOSSA is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with PYBOSSA.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+from flask import request, abort
+from flask_login import current_user
+from werkzeug.exceptions import BadRequest, Unauthorized, Forbidden
+from pybossa.api.api_base import APIBase
+from pybossa.model.project import Project
+from pybossa.model import DomainObject
+from pybossa.view.projects import project_by_shortname, get_locked_tasks
+
+class ProjectLocksAPI(APIBase):
+    """
+    Class for retreiving active locks in projects.
+
+    """
+    __class__ = Project
+
+    def _filter_query(self, repo_info, limit, offset, orderby):
+        if (len(request.args.keys()) == 0 or
+            (len(request.args.keys()) == 1 and "api_key" in request.args.keys())):
+            return []
+        if (not current_user.is_authenticated or
+             (not current_user.admin and not current_user.subadmin)):
+            raise Unauthorized("User not authorized for request")
```

Comment:
Are we okay with subadmins accessing project lock data for _all_ projects? If not, we need revise the implementation and that test case.

Reply:
Updated logic to verify subadmins are included as owner in project [here](https://github.com/bloomberg/pybossa/pull/933/commits/86953d171ad5e6bd1a93c8e144d76862270f1484#diff-3c29f6de870a6ac4fda6bf0364be63095558e9f759f281cb9c80302459a2603eR67).